tests/core/test_model_manager.py

```diff
@@ -529,3 +503,111 @@ def load(self, *args, **kwargs):
         dummy_model_path = os.path.join(TEST_DATA_PATH, DUMMY_BACKEND_MODEL_NAME)
         model = caikit.core.load(dummy_model_path)
         assert isinstance(model, DummyBaz)
+
+
+def test_load_must_return_model():
+    """Make sure that the return type of load is checked to be an instance of
+    ModuleBase, and will raise TypeError if it is not.
+    """
+
+    @caikit.core.block("00110203-baad-beef-0809-0a0b0c0d0e0f", "FunkyBlock", "0.0.1")
+    class _FunkyModel(SampleBlock):
+        @classmethod
+        def load(cls, model_path):
+            return (super().load(model_path), "something else")
+
+    model = _FunkyModel()
+    with tempfile.TemporaryDirectory() as tempdir:
+        # NOTE: the module will get detected as tests since _FunkyModel is defined here
+        model.save(tempdir)
+        with pytest.raises(TypeError):
+            caikit.core.load(tempdir)
+
+
+def test_load_with_new_shared_backend(good_model_path, reset_globals):
+    """If a shared laod backend has higher priority than LOCAL, it is used"""
+    loader_name = str(uuid.uuid4())
+    with temp_config(
+        {
+            "module_backends": {
+                "load_priority": [
+                    {"type": TestLoader.backend_type, "name": loader_name},
+                    {"type": backend_types.LOCAL},
+                ],
+            }
+        }
+    ):
+        configure()
+        model = caikit.core.load(good_model_path)
+        assert model.loader_name == loader_name
```

Comment:
I think this could be `model.get_load_backend.name == loader_name` and skip that extra `setattr`

Reply:
hot